app/models/chargeable_field.rb

```diff
@@ -136,8 +138,10 @@ def self.seed_data
   end
 
   def self.chargeable_cols_on_metric_rollup
-    existing_cols = MetricRollup.attribute_names
-    chargeable_cols = pluck(:metric) & existing_cols
-    chargeable_cols.map! { |x| VIRTUAL_COL_USES[x] || x }.sort
+    @chargeable_cols_on_metric_rollup ||= begin
+      existing_cols = MetricRollup.attribute_names
+      chargeable_cols = pluck(:metric) & existing_cols
```

Comment:
Could new records be created with new data in the `metric` column that would never be seen?

Reply:
Yes but not during running application. It could be added only by adding record (this happens very rarely) to [chargeable_field.yml](https://github.com/ManageIQ/manageiq/blob/59ad37fea6642a7a3e6beb31b007500a72c0c4cc/db/fixtures/chargeable_fields.yml) and based on the yaml, table chargeable_field table is filled up in the seed process.

thanks